webtau-standalone-runner-groovy/src/main/groovy/com/twosigma/webtau/runner/standalone/StandaloneTestRunner.groovy

```diff
@@ -47,16 +48,24 @@ class StandaloneTestRunner {
         currentTestPath = scriptPath.isAbsolute() ? scriptPath : workingDir.resolve(scriptPath)
 
         def relativeToWorkDirPath = workingDir.relativize(currentTestPath)
-        def script = groovy.createScript(relativeToWorkDirPath.toString(), new Binding())
 
-        script.setDelegate(delegate)
-        script.setProperty("scenario", this.&scenario)
-        script.setProperty("dscenario", this.&dscenario)
-        script.setProperty("sscenario", this.&sscenario)
-        script.setProperty("onlyWhen", this.&onlyWhen)
+        def scriptParse = new StandaloneTest(workingDir, currentTestPath, "parse/init", { ->
+            StandaloneTestListeners.beforeScriptParse(scriptPath)
+            def script = groovy.createScript(relativeToWorkDirPath.toString(), new Binding())
+
+            script.setDelegate(delegate)
+            script.setProperty("scenario", this.&scenario)
+            script.setProperty("dscenario", this.&dscenario)
+            script.setProperty("sscenario", this.&sscenario)
+            script.setProperty("onlyWhen", this.&onlyWhen)
 
-        StandaloneTestListeners.beforeScriptParse(scriptPath)
-        script.run()
+            script.run()
+        })
+
+        scriptParse.run()
+        if (scriptParse.hasError()) {
```

Comment:
May need to add check for assertion if anyone decides to add assertion outside of scenario.